website/content/docs/rfcs/cel-jwt.mdx

```diff
@@ -0,0 +1,177 @@
+# RFC: CEL Expression Support for JWT Authentication
+
+## Summary
+
+This feature offers support for [Common Expression Language (CEL)](https://github.com/google/cel-spec) in Vault's jwt-based auth engines (oidc, jwt) to create a flexible framework for validating claims and dynamically assigning token policies. CEL will allow administrators to define fine-grained rules for claim validation and dynamic policy assignment, extending the more static assignment of role- and identity-based policies.
+
+JWT CEL support is implemented as a new type of CEL role and follows the [general plan for CEL support](https://github.com/openbao/openbao/blob/main/website/content/docs/rfcs/cel-best-practices.mdx) in Openbao 
+
+---
+
+## Problem Statement
+
+Vault's jwt-based auth engines (OIDC and JWT) currently assemble token policies from `roles` and `identities`. However, administrators cannot enforce complex constraints, such as validating combinations of claims or dynamically assigning token policies based on claim values.
+
+---
+
+## User-facing description
+
+**_Administrators_** can define CEL roles under a new endpoint, `auth/{oidc|jwt}/cel/roles/:name`, to be executed during authentication. CEL roles evaluate incoming claims and generate `logical.Auth{}` instances having appropriate token policies.
+
+**_Users_** authenticating with JWT or OIDC tokens will have their claims validated through the CEL inidicated role. If successful, they receive a token with policies and other attributes. 
+
+For example:
+
+- A CEL role validates that the token contains specific claims like `"admin" in claims.roles && matches(claims.tenant_id, [0-9]+)"`.
+- If validation passes, the CEL program assigns the token policies `admin` and `team-12345` (using `claims.tenant_id` to form the policy name).
+
```

Comment:
Perhaps also give an example of such a program?

```suggestion

This could be codified in the following CEL program:

    "admin" in claims.roles && matches(claims.tenant_id, [0-9]+)" ? pb.Auth{ policies: ["admin", "team-12345"] } : false

```